score.rb

```diff
@@ -17,37 +17,65 @@ def initialize(frames)
 
   attr_accessor :frames
 
+  def single_frame_score(frame)
+    return frame.rolls.sum if frame.tenth_frame?
 
-  def single_frame_score(current_frame)
-    return current_frame.rolls.sum if current_frame.tenth_frame?
+    score_frame_type.each do |frame_type, score_frame|
+      return score_frame.call(frame) if frame_type?(frame_type, frame)
+    end
+  end
 
-    throws = current_frame.rolls
+  # TODO: is this method name descriptive enough?
+  # Method checks if the frame type to score matches the frame type in score_frame_type
+  def frame_type?(frame_type, frame)
+    send(frame_type, frame.rolls)
+  end
 
-    case
-    when strike?(throws.first) then score_strike(current_frame)
-    when spare?(throws) then score_spare(current_frame)
-    when open?(throws) then throws.sum
-    end
+  def score_frame_type
+    # Methods wrapped in Proc
+    {
+      strike?: Proc.new { |frame| score_strike(frame) },
+      spare?: Proc.new { |frame| score_spare(frame) } ,
+      open?: Proc.new { |frame| score_open(frame) }
+    }
+
+    # Methods wrapped in Lambda
+    {
+      strike?: ->(frame) { score_strike(frame) },
+      spare?: ->(frame) { score_spare(frame) },
+      open?: -> (frame) { score_open(frame) }
+    }
+
+    # Methods as object
+    {
+      strike?: method(:score_strike),
+      spare?: method(:score_spare),
+      open?: method(:score_open)
+    }
```

Comment:
I find this the most concise but not quite as readable as the lambda approach. 

It's clear which methods are being called but less clear that frame is passed to the respective method.

I am still reading up on use cases and when to/not to use this kind of approach.